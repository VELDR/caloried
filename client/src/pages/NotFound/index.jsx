import { FormattedMessage } from 'react-intl';
import { useNavigate } from 'react-router-dom';

import logo from '@static/images/not-found.svg';

import classes from './style.module.scss';

const NotFound = () => {
  const navigate = useNavigate();
  return (
    <div className={classes.contentWrapper} data-testid="not-found">
      <img className={classes.image} src={logo} alt="Not Found" />
      <div className={classes.title}>
        <FormattedMessage id="app_not_found" />
      </div>
      <div className={classes.button} onClick={() => navigate(-1)} data-testid="not-found-back">
        <FormattedMessage id="app_back" />
      </div>
    </div>
  );
};

export default NotFound;
