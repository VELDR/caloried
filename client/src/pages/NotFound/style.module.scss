@import '@styles/base';

.contentWrapper {
  @include flex-column(1rem);
  padding: 1rem;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: calc(100vh - 3.5rem);

  .image {
    @include box(clamp(20rem, 40vw, 50%), auto);
  }

  .title {
    font-size: 1.25rem;
    font-weight: 700;
    max-width: 30rem;
    text-align: center;
  }

  .button {
    @include primary-button;
  }
}
