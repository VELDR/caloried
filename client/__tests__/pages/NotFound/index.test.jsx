import NotFound from '@pages/NotFound';
import { fireEvent, render } from '@utils/testHelper';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let wrapper;

beforeEach(() => {
  wrapper = render(<NotFound />);
});

describe('NotFound page', () => {
  it('should render the page correctly', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('not-found')).toBeInTheDocument();
  });

  it('should call navigate back when back button is clicked', () => {
    const { getByTestId } = wrapper;

    const backButton = getByTestId('not-found-back');
    fireEvent.click(backButton);

    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
