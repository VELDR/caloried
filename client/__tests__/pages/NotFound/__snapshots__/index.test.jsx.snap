// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contentWrapper"
        data-testid="not-found"
      >
        <img
          alt="Not Found"
          class="image"
        />
        <div
          class="title"
        >
          Oh tidak! üôà Halaman ini sedang berpuasa. üçΩÔ∏è
        </div>
        <div
          class="button"
          data-testid="not-found-back"
        >
          Kembali
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contentWrapper"
      data-testid="not-found"
    >
      <img
        alt="Not Found"
        class="image"
      />
      <div
        class="title"
      >
        Oh tidak! üôà Halaman ini sedang berpuasa. üçΩÔ∏è
      </div>
      <div
        class="button"
        data-testid="not-found-back"
      >
        Kembali
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
